include/boost/fiber/detail/config.hpp

```diff
@@ -59,7 +59,48 @@
 // ARM Cortex-A15 32/64byte, Cortex-A9 16/32/64bytes
 // MIPS 74K: 32byte, 4KEc: 16byte
 // ist shoudl be safe to use 64byte for all
-static constexpr std::size_t cache_alignment{ 64 };
-static constexpr std::size_t cacheline_length{ 64 };
+# define CACHE_ALIGNMENT 64
+# define CACHELINE_LENGTH 64
+
+#if defined(BOOST_NO_CXX11_ALIGNAS)
+#  define cache_alignment CACHE_ALIGNMENT
+#  define cacheline_length CACHELINE_LENGTH
+#  define BOOST_FIBER_ALIGNAS_BEGIN(alignment)
+#  define BOOST_FIBER_ALIGNAS_END(alignment) __attribute__((aligned (alignment)))
```

Comment:
I don't think you should just assume that if no C++11 alignas that __attribute__aligned is available. i.e. These kinds of implementation specific definitions should be within the detection for the right implementation, e.g. #if defined(BOOST_GCC) && (BOOST_GCC > version)

But specifically for this case, when no C++11 alignas is available, why can't you just use BOOST_ALIGNMENT (provided by Boost.Config) which has meaningful definitions for multiple C++ implementations?

Reply:
On the other hands Boost.Fiber requires at least C++11, a 10-year old version of C++...
Does this still matter?